files/scripts/syncd.sh

```diff
@@ -105,6 +105,11 @@ function stopplatform1() {
     fi
 
     if [[ x$sonic_asic_platform != x"mellanox" ]] || [[ x$TYPE != x"cold" ]]; then
+        # Invoke platform specific pre shutdown routine.
+        PLATFORM=`$SONIC_DB_CLI CONFIG_DB hget 'DEVICE_METADATA|localhost' platform`
+        PLATFORM_PRE_SHUTDOWN="/usr/share/sonic/device/$PLATFORM/plugins/syncd_pre_shutdown.sh"
+        [ -f $PLATFORM_PRE_SHUTDOWN ] && $PLATFORM_PRE_SHUTDOWN start $DEV
```

Comment:
can we make change to have script within syncd and then execute from syncd context. for exampe:  docker exec -i syncd0 /usr/share/sonic/device/plugins/pre_shutdown.sh.  

This way we need not have vendor to force creating script to take start and $DEV as argument

Reply:
Added the script as part of vendor syncd docker.
/usr/bin/syncd_request_pre_shutdown which vendor may provide from syncd docker.

Reply:
@anamehra what if given vendor do not have this script ?

Reply:
if the file is not present, docker throws error which we are redirecting to /dev/null. In case the file is present, error messaging shall be handled by the vendor process.

 docker exec -it syncd0 /usr/bin/test
OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "/usr/bin/test": stat /usrbin/test: no such file or directory: unknown